packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart

```diff
@@ -71,6 +71,39 @@ class FirebaseAuth extends FirebasePluginPlatform {
     return null;
   }
 
+  /// Changes this instance to point to an Auth emulator running locally.
+  ///
+  /// Set the [origin] of the local emulator, such as "http://localhost:9099"
+  ///
+  /// Note: Must be called immediately, prior to accessing auth methods.
+  /// Do not use with production credentials as emulator traffic is not encrypted.
+  Future<void> useEmulator(String origin) async {
+    assert(origin.isNotEmpty);
+    String mappedOrigin = origin;
+
+    // Android considers localhost as 10.0.2.2 - automatically handle this for users.
+    if (defaultTargetPlatform == TargetPlatform.android) {
+      if (mappedOrigin.startsWith('http://localhost')) {
```

Comment:
It may replace slightly more cases than we want but at the same time the port is required and the regex below is both strict, throws an Exception with an explicit message that will fail immediately when attempted, and is coupled with documentation that gives a concrete / correct example. I'm comfortable with it.

Everyone wants to change the API here, but it's been discussed. Current idea that has rough consensus is that we emulate the firebase-js-sdk and take a String origin. If you feel strongly about this please talk to who ever has API design authority to make an authoritative decision and comment here with same. In absence of that, I will match firebase-js-sdk and take a string